api/ws/rpc/connect.js

```diff
@@ -56,6 +56,15 @@ const connectSteps = {
 	},
 
 	addSocket: peer => {
+		if (peer.socket) {
```

Comment:
if socket exist, does it even need to go through all the process?

Reply:
The socket can exist in a closed state and not destroyed, so every time the socket connection closes a new socket connection is issued from peers_manager with a different header(different height, broadhash) this will lead to multiple socket connection for same peer and these connections are not destroyed and leads to memory leaks.